signalapp/Signal-iOS
# Remove a strange artifact that appears to the right of a reaction to a message

### Contributor checklist

 * iPhone 11, iOS 13.5
 * iPhone 12, iOS 12.4

- - - - - - - - - -

### Description


The bug was originally reported in #4546 .

This strange behavior is was due to a bug in `UIStackView` that behaves strangely when its subviews are hidden and then shown. There are a few other places that has already handled this bug by applying `alpha` value to the hidden view. The code of the PR is to apply the trick to the reaction views too.

Note that a similar issue is fixed in #4637 but in a different way; this PR uses the `hide by alpha=0` trick. It works for this PR because it does not involve stretching subviews. For the other PR it had to remove and insert subviews because it involves stretching subviews in the `UIStackView`.

### Steps to verify the fix of the issue
It's probably easier to see the gif below to understand the steps.
1. Have a chat long enough so you must scroll to reach the bottom of it
1. Add an emoji reaction to a message that you receive, that is a few messages before the bottom of the chat
1. Scroll up the chat so the message that has the reaction emoji is the next one to appear on screen
1. Go back to the conversation list screen
1. Come back to the chat again
1. Scroll down to the bottom

**Actual behavior before the fix**

A strange artifact appears to the right of the emoji reaction.

**Expected behavior after the fix**

The artifact does not appear.

### Screenshots

|Before|After|
|---|---|
|![artifact-bug](https://user-images.githubusercontent.com/28482/100287703-1e253000-2f43-11eb-9994-d4cf392542aa.png)|![artifact-fixed](https://user-images.githubusercontent.com/28482/100287701-1e253000-2f43-11eb-9453-291ee2ac8f7c.png)|
|![artifact-bug-ios12](https://user-images.githubusercontent.com/28482/100287761-44e36680-2f43-11eb-9a92-491f59da2edd.gif)|![artifact-fixed-ios12](https://user-images.githubusercontent.com/28482/100287760-444ad000-2f43-11eb-8dd8-2a925f5ed10d.gif)|
